src/psp/msg_report.rs

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (C) 2023 IBM
+ *
+ * Authors:
+ *   Claudio Carvalho <cclaudio@linux.ibm.com>
+ */
+
+use crate::prints;
+
+use alloc::boxed::Box;
+use core::slice;
+
+/// SnpReportRequest size
+const REQUEST_SIZE: usize = core::mem::size_of::<SnpReportRequest>();
+
+/// 64
+pub const USER_DATA_SIZE: usize = 64;
+
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone)]
+pub struct SnpReportRequest {
+    user_data: [u8; USER_DATA_SIZE],
+    vmpl: u32,
+    rsvd: [u8; 28usize],
+}
+
+impl SnpReportRequest {
+    pub fn new() -> Self {
+        Self {
+            user_data: [0u8; 64],
+            vmpl: 0u32,
+            rsvd: [0u8; 28],
+        }
+    }
+
+    pub fn set_user_data(&mut self, data: &[u8; USER_DATA_SIZE]) {
+        self.user_data.copy_from_slice(data);
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
+        unsafe { slice::from_raw_parts(self as *const _ as *const u8, REQUEST_SIZE) }
+    }
+}
+
+#[repr(C)]
+#[repr(align(2048))]
+#[derive(Debug, Copy, Clone)]
+pub struct SnpReportResponse {
+    status: u32,
+    report_size: u32,
+    reserved: [u8; 24usize],
+    report: AttestationReport,
+}
```

Comment:
I actually need to read these fields, so please:

```suggestion
pub struct SnpReportResponse {
    pub status: u32,
    pub report_size: u32,
    _reserved: [u8; 24],
    pub report: AttestationReport,
}
```

Reply:
For code consistency, better to use getters and setters (see `funcs!()`). If they don't read them in this PR, no need to include it within their commits